src/main/java/com/pinterest/secor/uploader/HadoopS3UploadManager.java

```diff
@@ -60,12 +61,23 @@ public HadoopS3UploadManager(SecorConfig config) {
         LOG.info("uploading file {} to {}", localLogFilename, logFileName);
 
         final Future<?> f = executor.submit(new Runnable() {
+        	final int totalRetries = Integer.parseInt(mConfig.getUploaderRetries());
             @Override
             public void run() {
+            	int retriesLeft = totalRetries;
+            	boolean success = false;
+            	while (retriesLeft > 0 && !success) {
                 try {
                     FileUtil.moveToCloud(localLogFilename, logFileName);
+            			success = true;
                 } catch (IOException e) {
-                    throw new RuntimeException(e);
+            			success = false;
+            			retriesLeft--;
+            			if (retriesLeft > 0)
+            				LOG.warn("Error uploading files to s3. Will retry {} times. Exception {}", retriesLeft, e);
+            			else 
+            				Throwables.propagate(e);
```

Comment:
Looks like Throwables.propage is no longer recommended: https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate